packages/evm/src/fungible.ts

```diff
@@ -200,4 +214,27 @@ class EvmFungibleAssetTransfer extends EvmTransfer {
 
     return createTransactionRequest(transferTx);
   }
+
+  async verifyAccountBalance(fee: EvmFee): Promise<void> {
+    const resource = this.resource as EvmResource;
+    const provider = new Web3Provider(this.sourceNetworkProvider);
+
+    // Native Token Balance check
+    if ([FeeHandlerType.BASIC, FeeHandlerType.TWAP].includes(fee.type)) {
+      const nativeBalance = await provider.getBalance(this.sourceAddress);
+
+      if (nativeBalance.lt(fee.fee))
+        throw new Error(`Insufficient native token balance for network ${this.sourceDomain.name}`);
+    }
+
+    // ERC20 Token Balance check
+    if ([FeeHandlerType.BASIC, FeeHandlerType.PERCENTAGE].includes(fee.type)) {
```

Comment:
```typescript
    if (FeeHandlerType.PERCENTAGE === fee.type) {
```

Reply:
Here we apply for both scenarios if it's Basic and Percentage

Reply:
in `BASIC` fee is not deducted in `erc20` so its not needed